caffe2/CMakeLists.txt

```diff
@@ -211,10 +206,579 @@ if(NOT BUILD_ATEN_ONLY)
+  if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    message(STATUS "${CMAKE_CXX_COMPILER} ${TORCH_SRC_DIR}/abi-check.cpp -o ${CMAKE_BINARY_DIR}/abi-check")
+    execute_process(
+      COMMAND
+      "${CMAKE_CXX_COMPILER}"
+      "${TORCH_SRC_DIR}/abi-check.cpp"
+      "-o"
+      "${CMAKE_BINARY_DIR}/abi-check"
+      RESULT_VARIABLE ABI_CHECK_COMPILE_RESULT)
+    if (ABI_CHECK_COMPILE_RESULT)
+      message(FATAL_ERROR "Could not compile ABI Check: ${ABI_CHECK_COMPILE_RESULT}")
+    endif()
+    execute_process(
+      COMMAND "${CMAKE_BINARY_DIR}/abi-check"
+      RESULT_VARIABLE ABI_CHECK_RESULT
+      OUTPUT_VARIABLE GLIBCXX_USE_CXX11_ABI)
+    if (ABI_CHECK_RESULT)
+      message(WARNING "Could not run ABI Check: ${ABI_CHECK_RESULT}")
+    endif()
+    message(STATUS "Determined _GLIBCXX_USE_CXX11_ABI=${GLIBCXX_USE_CXX11_ABI}")
+  endif()
+
+  # CMake config for external projects.
+  configure_file(
+    ${PROJECT_SOURCE_DIR}/cmake/TorchConfigVersion.cmake.in
+    ${PROJECT_BINARY_DIR}/TorchConfigVersion.cmake
+    @ONLY)
+  configure_file(
+    ${TORCH_ROOT}/cmake/TorchConfig.cmake.in
+    ${PROJECT_BINARY_DIR}/TorchConfig.cmake
+    @ONLY)
+  install(FILES
+    ${PROJECT_BINARY_DIR}/TorchConfigVersion.cmake
+    ${PROJECT_BINARY_DIR}/TorchConfig.cmake
+    DESTINATION share/cmake/Torch)
+
+  if (USE_DISTRIBUTED)
+    add_subdirectory(${TORCH_SRC_DIR}/lib/THD lib_THD)
+    if (NOT MSVC AND NOT APPLE)
+      add_subdirectory(${TORCH_SRC_DIR}/lib/c10d lib_c10d)
+    endif()
+  endif()
+
+
+  # ---[ Torch python bindings build
+  add_subdirectory(../torch torch)
+
+  
+endif()
+# ==========================================================
+# END formerly-libtorch flags
+# ==========================================================
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+if (NOT NO_API)
+  target_include_directories(caffe2 PUBLIC
+    $<BUILD_INTERFACE:${TORCH_SRC_DIR}/csrc/api>
+    $<BUILD_INTERFACE:${TORCH_SRC_DIR}/csrc/api/include>)
+endif()
+
+
+# kostmo
+#if (MSVC)
+#  target_link_libraries(caffe2 PUBLIC onnx onnx_library)
+#endif()
+
+
+find_package(OpenMP QUIET)
+if(OPENMP_FOUND)
+  message(STATUS "pytorch is compiling with OpenMP. \n"
+    "OpenMP CXX_FLAGS: ${OpenMP_CXX_FLAGS}. \n"
+    "OpenMP libraries: ${OpenMP_CXX_LIBRARIES}.")
+  target_compile_options(caffe2 INTERFACE ${OpenMP_CXX_FLAGS})
+  target_link_libraries(caffe2 PUBLIC ${OpenMP_CXX_LIBRARIES})
+endif()
+
+
+
+
+if(USE_ROCM)
+
+  # XXX kostmo
+#  target_link_libraries(caffe2 PUBLIC caffe2_hip_library)
+
+  target_compile_definitions(caffe2 PRIVATE
+    USE_ROCM
+    __HIP_PLATFORM_HCC__
+    )
+  target_include_directories(caffe2 PRIVATE
+    /opt/rocm/include
+    /opt/rocm/hcc/include
+    /opt/rocm/rocblas/include
+    /opt/rocm/hipsparse/include
+    )
+endif()
+
+
+
+
+
 if (NOT WIN32)
-  target_compile_options(caffe2 PRIVATE "-fvisibility=hidden")
+    # FIXME kostmo
+#  target_compile_options(caffe2 PRIVATE "-fvisibility=hidden")
```

Comment:
after we fixed THPPointer - can we bring it on or are there still a lot of failures?

Reply:
There are still failures, e.g.:

    ImportError: /scratch/kostmo/anaconda3/lib/python3.7/site-packages/torch/lib/libtorch_python.so: undefined symbol: _ZTIN5torch3jit4NodeE